code/starter_script - Copy.R

```diff
@@ -0,0 +1,62 @@
+# an example of paired predictions and observations
+library(dplyr)
+dat <- readRDS('data/predicted_observed_temps.rds')
+head(dat)
+
+# note that SNTemp predictions (uncalibrated process model)
+# have predictions for all places/times
+# here you just have the test period from the hybrid model (rgcn2_full_temp_c)
+# and sparse observations (some reaches measured, some not)
+
+summary(dat)
+str(dat)
+# example RMSE calculation
+# https://github.com/USGS-R/delaware-model-prep/blob/master/3_predictions/src/munge_predictions.R#L103-L113
+
+# removing rows with na values. 
+rgnc_dat_filter <- dat %>% filter(!is.na(temp_c), !is.na(rgcn2_full_temp_c))
+str(rgnc_dat_filter)
+summary(rgnc_dat_filter)
+rgnc_by_seg <- rgnc_dat_filter %>% group_by(seg_id_nat)
+## Absolute Residual Error: calculating Mean Absolute Error (MAE) metric to compare both models prediction data vs observed data. We found the mae by finding the sum of the absolute value difference in predicted and observed temperature. Then divided the sum by the n row in grouped data.
+
+cal_mae <- function(observe_data, predict_data) {
+    mae = (mean(abs(observe_data - predict_data), na.rm = TRUE ))
+    return(mae)
+}
+
+mae_process_model <- rgnc_by_seg %>% 
+  summarize(MAE_Process_Model = round(cal_mae(temp_c, sntemp_temp_c), 2))
```

Comment:
I think you should put the round in the function, so you don't have to repeat it. You could even have number of digits as an input to the function, with a default of 2, but with the flexibility to change.

Reply:
Additionally, you could simplify some of this code by doing two summarizes in the pipeline, and calling the new data `mae` or something like: 
```
mae <- rgnc_by_seg %>% 
  summarize(MAE_Process_Model = round(cal_mae(temp_c, sntemp_temp_c), 2),
                     MAE_Hybrid_Model = round(cal_mae(temp_c, rgcn2_full_temp), 2))
```